st2api/st2api/controllers/resource.py

```diff
@@ -119,6 +123,9 @@ def _get_all(self, exclude_fields=None, sort=None, offset=0, limit=None, query_o
         offset = int(offset)
 
         if limit and int(limit) > self.max_limit:
+            # TODO: We should throw here, I don't like this.
```

Comment:
As mentioned in the comment, I'm not fan of this "silently ignore invalid value" approach.

Imo, we should throw `ValueError` with appropriate message which results in 400 status code.

What do others think?